src/thor/bidirectional_astar.cc

```diff
@@ -516,16 +523,29 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
   Init(origin_new, destination_new);
 
   // Get time information for forward and backward searches
+  bool invariant = options.has_date_time_type() && options.date_time_type() == Options::invariant;
   auto forward_time_info = TimeInfo::make(origin, graphreader, &tz_cache_);
   auto reverse_time_info = TimeInfo::make(destination, graphreader, &tz_cache_);
 
+  // When a timedependent route is too long in distance it gets sent to this algorithm. It used to be
+  // the case that this algorithm called EdgeCost without a time component. This would result in
+  // timedependent routes falling back to time independent routing. Now that this algorithm is time
+  // aware we will be tracking time in one direction of the search. To revert to previous behavior
+  // you can uncomment the code below and get a time independent route in the fallback scenario.
+  //  if (!invariant) {
```

Comment:
this is an important change in what happens when we fail over from timedependent to bidir a* due to exceeding max length. please read!